Platform/obc/hal/comms_hal.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2015  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/**
+ * @file comms_hal.c
+ * @author Arash Yazdani
+ * @date 2020-10-01
+ */
+
+/*Once the hardware is connected, toggle this Macro*/
+//#define Stubbed = 1;
+
+#include "obc_hal.h"
+#include "comms_hal.h"
+#include "hal.h"
+#include "services.h"
+#include "queue.h"
+
+
+
+/*Comments for the PR reviewer:
+ * 2- Like RTC we can check if the value is stored after the
+ * set functions (change the get functions).
+ * But there is no point when eventually Stubbed will be 0.
+ * 3- Do we need comments for each function?
+ */
+
+STX_return HAL_S_getFreq (float *S_freq) {
+  #ifdef SBAND_IS_STUBBED
+    *S_freq = (float)2210.5;
+    return IS_STUBBED;
+  #else
+    return STX_getFrequency(*S_freq);
+  #endif
+};
+
+STX_return HAL_S_getControl (Sband_PowerAmplifier *S_PA) {
+  #ifdef SBAND_IS_STUBBED
+    S_PA->status = 1;
+    S_PA->mode = 3;
+    return IS_STUBBED;
+  #else
+    return STX_getControl(&S_PA->status, &S_PA->mode);
+  #endif
+};
+
+STX_return HAL_S_getEncoder (Sband_Encoder *S_Enc){
+  #ifdef SBAND_IS_STUBBED
+    S_Enc->scrambler = 1;
+    S_Enc->filter = 6;
+    S_Enc->modulation = 5;
+    S_Enc->rate = 4;
+    return IS_STUBBED;
+  #else
+    return STX_getEncoder&S_Enc->scrambler, &S_Enc->filter, &S_Enc->filter, &S_Enc->filter);
```

Comment:
This line won't compile

Reply:
Oops. I forgot the parenthesis and repeated the filter multiple times. That must be it.